aya/src/programs/links.rs

```diff
@@ -252,14 +252,23 @@ impl ProgAttachLink {
         prog_fd: BorrowedFd<'_>,
         target_fd: BorrowedFd<'_>,
         attach_type: bpf_attach_type,
+    ) -> Result<Self, ProgramError> {
+        Self::attach_with_flags(prog_fd, target_fd, attach_type, 0)
+    }
+
+    pub(crate) fn attach_with_flags(
```

Comment:
I'd prefer changing the API here to add `flags: u32` instead of adding a new funciton.

Reply:
Sure thing, done.